internetarchive/openlibrary
# Add loading and success indicators for backend checks

Closes #9590.


Feature. Adds loading and success indicator icons for API requests in registration form, i.e. to check if username/email is available. Avoids awkward pause as checks run and sudden error message appearance/disappearance when validating on key-up.

### Technical

Very simple! Ended up just going with two images, both initially rendered `display: none`, and then conditionally shown/hidden via the script in `realtime_account_validation.js`.

Changed a few class names around to be able to apply the password icon styling to all three, and to be able to query the correct icons to show/hide.

### Testing

1. Go to `/account/create`
2. Enter a non-taken username or email address and click onto the next input
3. A spinner should appear right away, replaced by a checkmark when the check is complete 
4. Try the same process but this time with a taken username or email address
5. A spinner should appear right away, and when the check is complete it should disappear and an error message should show as usual

### Screenshot

<img width="437" alt="Loading indicator in action" src="https://github.com/user-attachments/assets/3122ab0e-646b-4e36-9230-142e9013c989">
<img width="378" alt="Success indicator in action" src="https://github.com/user-attachments/assets/056a7a5f-9525-4a74-b33d-eb0ff0bc4745">


### Stakeholders

@cdrini
